include/onnxruntime/core/providers/tensorrt/tensorrt_provider_options.h

```diff
@@ -46,4 +46,7 @@ struct OrtTensorRTProviderOptionsV2 {
   const char* trt_profile_max_shapes{nullptr};           // Specify the range of the input shapes to build the engine with
   const char* trt_profile_opt_shapes{nullptr};           // Specify the range of the input shapes to build the engine with
   int trt_cuda_graph_enable{0};                          // Enable CUDA graph in ORT TRT
+  int trt_dump_ep_context_model{0};                      // Dump EP context node model
```

Comment:
fyi https://github.com/microsoft/onnxruntime/pull/18877
added these options at session level. so all EP's can use them.

Reply:
After taking a closer look, i think it requires more changes for TRT EP to consume session options.
QNN EP factory [creation](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/provider_registration.cc#L97) is in `OrtApis::SessionOptionsAppendExecutionProvider` where it passes session options to QNN EP factory constructor.
However, the current TRT EP factory [initialization](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/provider_bridge_ort.cc#L1696) does not consume session options. However, we can still make it work but requires more work that is not related to this "direct load" PR.

So, i kind of want to keep those TRT EP provider options as an option. We can have another PR to make TRT EP consume session options later.

Reply:
> ok, let's keep it separate for now. I just want to confirm the session option support can be done in time for ort 1.17. we don't want the situation where 1.17 supports trt ep option and then session option after the release.
It can be done for ORT 1.17 and i'm starting to work on this now